### PR TITLE
WIP

### PR DESCRIPTION
**Pull request by aucuparia**

### CODE REVIEW

**Review by lwesterhof — COMMENTED** [review]
LGTM

**Comment by aucuparia** [comment]
Rewrite of entire workflow is finished. Some polishing left, mostly Bootstrap stuff and code cleanup, but now would be a good time for first review.

File: datarequest/datarequest.py [hunk 1]
```diff
@@ -3,15 +3,422 @@
...
+
+@datarequest_bp.route('upload_dta/<request_id>', methods=['POST'])
+def upload_dta(request_id):
+    if not permission_check(request_id, ['DM'], ['PREREGISTRATION_CONFIRMED', 'DAO_APPROVED']):
+        abort(403)
+    
+    filename = secure_filename(request.files['file'].filename)
+    file_path = os.path.join("/" + g.irods.zone, 'home', 'datarequests-research', request_id, 'dta', filename)
+
+    # Verify that uploaded file is a PDF
```
Comment (lwesterhof) [inline datarequest/datarequest.py:336]: PDF verification is missing.

File: datarequest/datarequest.py [hunk 2]
```diff
@@ -3,15 +3,422 @@
...
 )
 
+# Helper functions
+
+def permission_check(request_id, roles, statuses):
+    return api.call('datarequest_action_permitted', {'request_id': request_id,
+                                                     'statuses': statuses,
+                                                     'roles': roles})['data']
+
+def human_readable_status(request_status):
```
Comment (lwesterhof) [inline datarequest/datarequest.py]: May be nicer and more readable to use a Enum, for example: https://github.com/UtrechtUniversity/yoda-ruleset/blob/a0e4cfcf2fb5c6c68902cac3a3967ee8211a017e/util/constants.py#L122

File: datarequest/datarequest.py [hunk 3]
```diff
@@ -3,15 +3,422 @@
...
+                           is_dac_member=is_dac_member,
+                           help_contact_name='Todo_help_contact_name',
+                           help_contact_email='Todo_help_contact_email')
+
+
+@datarequest_bp.route('view/<request_id>')
+def view(request_id):
+    roles = api.call('datarequest_roles_get', {'request_id': request_id})['data']
+
+    if not 'PM' in roles and not 'DM' in roles and not 'DAC' in roles and not 'OWN' in roles:
```
Comment (lwesterhof) [inline datarequest/datarequest.py]: If we define this condition after the is_* declarations, we have a more readable conditon.

File: datarequest/datarequest.py [hunk 4]
```diff
@@ -3,15 +3,422 @@
...
+
+@datarequest_bp.route('upload_attachment/<request_id>', methods=['POST'])
+def upload_attachment(request_id):
+    if not permission_check(request_id, ['OWN'], ['PENDING_ATTACHMENTS']):
+        abort(403)
+
+    filename = secure_filename(request.files['file'].filename)
+    file_path = os.path.join("/" + g.irods.zone, 'home', 'datarequests-research', request_id, 'attachments', filename)
+
+    api.call('datarequest_attachment_upload_permission', {'request_id': request_id, 'action': 'grant'})
```
Comment (lwesterhof) [inline datarequest/datarequest.py]: Try and catch these API calls to handle errors?

**Review by lwesterhof — CHANGES_REQUESTED** [review]

**Review by lwesterhof — APPROVED** [review]
LGTM

**Pull request merged**